### PR TITLE
Fix persistence issues

### PR DESCRIPTION
**Pull request by etorreborre**
This PR fixes a number of regressions after the merge of the [database persistence PR](https://github.com/build-trust/ockam/pull/6913):

 - incorrect use of the space id instead of the space name in a command
 - create the authority identity for the identity node if it doesn't exist
 - correctly stop nodes in the bats test suite when running concurrently
 - create a default identity only once when using the function accepting optional values

### CODE REVIEW

File: implementations/rust/ockam/ockam_identity/src/error.rs [hunk 1]
```diff
@@ -1,17 +1,19 @@
+use ockam_core::compat::string::String;
 use ockam_core::{
     errcode::{Kind, Origin},
     Error,
 };
 
 /// Identity crate error
+#[repr(u8)]
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_identity/src/error.rs:8]: Why is that required?

**Review by SanjoDeundiak — DISMISSED** [review]

File: implementations/rust/ockam/ockam_identity/src/error.rs (hunk 1 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_identity/src/error.rs:8]: Because I have added a `String` to the `InvalidIdentifier` error.

File: implementations/rust/ockam/ockam_api/src/cli_state/identities.rs [hunk 2]
```diff
@@ -49,17 +49,40 @@ impl CliState {
...
+                    .await
+            }
+            None => {
+                let default_identity = self
+                    .identities_repository()
+                    .await?
+                    .get_default_named_identity()
+                    .await?;
+                if let Some(default_identity) = default_identity {
+                    if default_identity.vault_name == vault_name {
```
Comment (davide-baldo) [inline implementations/rust/ockam/ockam_api/src/cli_state/identities.rs]: What if the user didn't provide a vault name? I don't think we need to check that the default user belongs to the default vault.

Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/cli_state/identities.rs]: I can indeed skip this check. At a fundamental level we have an issue with our command line options and actions. We can define a default identity, a default vault, other identities and vaults. Then we can use `ockam identity default` and `ockam vault default` to change either the default identity or the default vault so that they end-up being inconsistent with each other.

I think that the proper fix is to separate the domains and never allow to pass a `--vault` option when `--identity` is specified. And if we pass `--identity` we always expect it to exist.

Comment (davide-baldo) [reply implementations/rust/ockam/ockam_api/src/cli_state/identities.rs]: I agree, passing vault when we are already passing an identity is confusing. I like your proposal of mutually exclude vaults and identities. The proposed solution would also prevent us from lazily creating identities. A middle ground could be change `--vault` with `--vault-when-creating-identity` (maybe with better wording: )

Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/cli_state/identities.rs]: I have now simply removed that function which was just used in 2 places so we don't get to deal with all the combinations.

**Review by SanjoDeundiak — DISMISSED** [review]

**Review by davide-baldo — DISMISSED** [review]

**Review by polvorin — APPROVED** [review]
